flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java

```diff
@@ -350,14 +350,48 @@ public KubernetesLeaderElector createLeaderElector(
     @Override
     public CompletableFuture<Void> deleteConfigMapsByLabels(Map<String, String> labels) {
         return CompletableFuture.runAsync(
-                () -> this.internalClient.configMaps().withLabels(labels).delete(),
+                () -> {
+                    if (!this.internalClient.configMaps().withLabels(labels).delete()) {
```

Comment:
The current implementation assumes that a normal response(`code >= 200 && code < 300`) means the K8s deleted the ConfigMaps successfully. So we also should believe that a `404` code means the ConfigMap does not exist, maybe deleted externally.

Do we really need to check again the existence when the return value of `delete()` is `false`? IIUC, both the `true` and `false` indicate the ConfigMaps does not exist now.

Reply:
I couldn't find anything about the actual semantics of the 404 in the k8s API reference (see [ConfigMap DELETE for k8s 1.23](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#delete-configmap-v1-core).

But based on the meaning of 404 being Not Found, it's quite likely that it means that the resource didn't exist in the first place. The thing is that this is an implementation detail of the Fabric8Client implementation. Our implementation should focus on the contract presented in the interface that is used (in our case, the Fabric8 client's [Deletable](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#delete-configmap-v1-core) interface. And its JavaDoc doesn't specify the semantics behind the `false` value.

Relying on the contract described in the JavaDoc, we still would need to do the existence check. But I agree, that implementation-wise it's not an issue, I guess, looking into how the [fabric8's BaseOperation.delete](https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/BaseOperation.java#L485) is implemented.

Reply:
IMHO, if the exist check is expensive, we shouldn't do it and rather rely on the implementation but add a comment in the `FlinkKubeClient.deleteConfigMapsBy*` implementations. Doing the exist call would just be more coherent with the actual contract of the interface provided by the fabric8 client.

Reply:
It seems that Kubernetes APIServer is using [go-rest](https://github.com/emicklei/go-restful) for rest handling. From the [documentation](https://pkg.go.dev/github.com/emicklei/go-restful#hdr-Error_Handling), the 404 will be returned if the path could not be found(e.g. /api/v1/namespaces/test/configmaps/test-config-map).

So in the fabric8 K8s client implementation, only throwing exception indicates that the deletion operation might fail. I am suggesting to add a comment to explain why it could work.